nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-db-schema-registry-service/src/main/java/org/apache/nifi/db/schemaregistry/DatabaseSchemaRegistry.java

```diff
@@ -0,0 +1,179 @@
+    private volatile String dbSchemaName;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propDescriptors;
+    }
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        dbcpService = context.getProperty(DBCP_SERVICE).asControllerService(DBCPService.class);
+        dbCatalogName = context.getProperty(CATALOG_NAME).evaluateAttributeExpressions().getValue();
+        dbSchemaName = context.getProperty(SCHEMA_NAME).evaluateAttributeExpressions().getValue();
+    }
+
+    @Override
+    public RecordSchema retrieveSchema(SchemaIdentifier schemaIdentifier) throws IOException, SchemaNotFoundException {
+        if (schemaIdentifier.getName().isPresent()) {
+            return retrieveSchemaByName(schemaIdentifier);
+        } else {
+            throw new SchemaNotFoundException("This Schema Registry only supports retrieving a schema by name.");
+        }
+    }
+
+    @Override
+    public Set<SchemaField> getSuppliedSchemaFields() {
+        return schemaFields;
+    }
+
+    RecordSchema retrieveSchemaByName(final SchemaIdentifier schemaIdentifier) throws IOException, SchemaNotFoundException {
+        final Optional<String> schemaName = schemaIdentifier.getName();
+        if (schemaName.isEmpty()) {
+            throw new SchemaNotFoundException("Cannot retrieve schema because Schema Name is not present");
+        }
+
+        final String tableName = schemaName.get();
+        try {
+            try (final Connection conn = dbcpService.getConnection()) {
+                final DatabaseMetaData dmd = conn.getMetaData();
+                try (final ResultSet colrs = dmd.getColumns(dbCatalogName, dbSchemaName, tableName, "%")) {
+                    final List<RecordField> recordFields = new ArrayList<>();
+                    while (colrs.next()) {
+                        // COLUMN_DEF must be read first to work around Oracle bug, see NIFI-4279 for details
+                        final String defaultValue = colrs.getString("COLUMN_DEF");
+                        final String columnName = colrs.getString("COLUMN_NAME");
+                        final int dataType = colrs.getInt("DATA_TYPE");
+                        final String nullableValue = colrs.getString("IS_NULLABLE");
+                        final boolean isNullable = "YES".equalsIgnoreCase(nullableValue) || nullableValue.isEmpty();
+                        recordFields.add(new RecordField(
+                                columnName,
+                                DataTypeUtils.getDataTypeFromSQLTypeValue(dataType),
+                                defaultValue,
+                                isNullable));
+                    }
+
+                    // If no columns are found, check that the table exists
+                    if (recordFields.isEmpty()) {
+                        try (final ResultSet tblrs = dmd.getTables(dbCatalogName, dbSchemaName, tableName, null)) {
```

Comment:
Recommend adjusting the variable name for readability:
```suggestion
                        try (final ResultSet tablesResultSet = dmd.getTables(dbCatalogName, dbSchemaName, tableName, null)) {
```